src/Options/OptionsTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Google\ApiCore\Options;
+
+use Google\ApiCore\ValidationException;
+use BadMethodCallException;
+
+trait OptionsTrait
+{
```

Comment:
nit: maybe we move this to `Google\ApiCore\OptionsTrait`? That way the `Options` dir is only for client options.

Reply:
Just a thought but I think this dir seems the right place, as it's usage is in this directory _only_.

Perhaps, `Google\ApiCore\Options\Traits` can work too.